gobblin-core/src/main/java/gobblin/writer/PartitionedDataWriter.java

```diff
@@ -171,4 +173,35 @@ public void close() throws IOException {
     return this.builder.get().forPartition(partition).withWriterId(this.baseWriterId + "_" + this.writerIdSuffix++)
         .build();
   }
+
+  @Override
+  public State getFinalState() {
+
+    State state = new State();
+    try {
+      for (Map.Entry<GenericRecord, DataWriter<D>> entry : this.partitionWriters.asMap().entrySet()) {
+        if (entry.getValue() instanceof FinalState) {
+
+          State partitionFinalState = ((FinalState) entry.getValue()).getFinalState();
+
+          if (this.shouldPartition) {
+            for (String key : partitionFinalState.getPropertyNames()) {
+              // Prevent overwriting final state across writers
+              partitionFinalState.setProp(key + "_" + AvroUtils.serializeAsPath(entry.getKey(), false, true),
+                  partitionFinalState.getProp(key));
+            }
+          }
+
+          state.addAll(partitionFinalState);
+        }
+      }
+      state.setProp("RecordsWritten", recordsWritten());
+      state.setProp("BytesWritten", bytesWritten());
```

Comment:
Makes sense if it's been there before. I thought this is something you just added.